src/routes/relay/entities/schema/sponsored-call.schema.helper.ts

```diff
@@ -166,3 +171,90 @@ export const isValidMultiSendCall = (
 
   return true;
 };
+
+// ===================== createProxyWithNonce ======================
+
+export const isCreateProxyWithNonceCalldata = (data: string): boolean => {
+  const SETUP_TX_SIGNATURE = 'createProxyWithNonce(address,bytes,uint256)';
+
+  return isCalldata(data, SETUP_TX_SIGNATURE);
+};
+
+interface CreateProxyWithNonceTransactionData {
+  readonly singleton: string;
+  readonly initializer: string;
+  readonly saltNonce: bigint;
+}
+
+const decodeCreateProxyWithNonce = (
+  encodedData: string,
+): CreateProxyWithNonceTransactionData => {
+  const CREATE_PROXY_WITH_NONCE_FRAGMENT =
+    'function createProxyWithNonce(address _singleton, bytes memory initializer, uint256 saltNonce)';
+
+  const createProxyWithNonceInterface = new ethers.Interface([
+    CREATE_PROXY_WITH_NONCE_FRAGMENT,
+  ]);
+
+  const [singleton, initializer, saltNonce] =
+    createProxyWithNonceInterface.decodeFunctionData(
+      CREATE_PROXY_WITH_NONCE_FRAGMENT,
+      encodedData,
+    );
+
+  return {
+    singleton,
+    initializer,
+    saltNonce,
+  };
+};
+
+export const isValidCreateProxyWithNonceCall = (
+  chainId: string,
+  to: string,
+  data: string,
+): boolean => {
+  if (!isCreateProxyWithNonceCalldata(data)) {
+    return false;
+  }
+
+  const proxyFactoryDeployment = getProxyFactoryDeployment({
+    network: chainId,
+  });
+
+  if (!proxyFactoryDeployment || to !== proxyFactoryDeployment.defaultAddress) {
+    return false;
+  }
+
+  const { singleton } = decodeCreateProxyWithNonce(data);
```

Comment:
If we pass the `proxyFactoryDeployment` to `decodeCreateProxyWithNonce` we could create the Interface from the abi included in our deployment:
```ts
new Interface(proxyFactoryDeployment.abi)
```

Same goes for the the other Safe Interfaces.

Reply:
I like this but `decodeCreateProxyWithNonce` is not always called within functions that have the interface. I'll leave this as is for now.